python/ray/data/tests/test_auto_parallelism.py

```diff
@@ -21,6 +21,30 @@ def test_auto_parallelism_basic(shutdown_only):
     assert ds.num_blocks() == 150, ds
 
 
+def test_auto_parallelism_parquet(shutdown_only, tmp_path):
```

Comment:
Instead of testing the entire auto parallelism logic, could we unit test just the size_bytes() / estimate_inmemory_data_size() of the ParquetReader directly? That would be less brittle to changes.

Reply:
@ericl - make sense, updated to test `_ParquetDatasourceReader.estimate_inmemory_data_size` only.